pymodbus/logging.py

```diff
@@ -32,12 +32,15 @@ class Log:
     _logger = logging.getLogger(__name__)
 
     @classmethod
-    def apply_logging_config(cls, level=logging.WARNING):
-        """Apply basic logging configuration"""
-        logging.basicConfig(
-            format="%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
-            datefmt="%H:%M:%S",
-        )
+    def apply_logging_config(
+        cls, level: int = logging.WARNING, **config_overrides: dict
+    ) -> None:
+        """Apply basic logging configuration."""
+        default_config: dict = {
+            "format": "%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
+            "datefmt": "%H:%M:%S",
```

Comment:
please add **config_overrides here.

Reply:
```python
default_config: dict = {
    **config_overrides,
    "format": "%(asctime)s %(levelname)-5s %(module)s:%(lineno)s %(message)s",
    "datefmt": "%H:%M:%S",
}
```

I think adding it here suggests the overrides are part of the default, so I would prefer not to do that.

Reply:
it is only a local variable, so feel free to rename it to e.g. effective_config